src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/MethodBodyScanner.cs

```diff
@@ -762,17 +759,29 @@ Stack<StackSlot> currentStack
             }
         }
 
-        private static void ScanLdtoken(MethodIL methodBody, TypeDesc type, Stack<StackSlot> currentStack)
+        private static void ScanLdtoken(MethodIL methodBody, object operand, Stack<StackSlot> currentStack)
         {
-            if (type.IsGenericParameter)
+            if (operand is TypeDesc type)
```

Comment:
Nit: You can use the new shine pattern matching in cases like this:

switch (operand)
{
    case TypeDesc type:
    ...
    case MethodDesc method:
    ...
    default:
    ....
}
```